python/llm/src/ipex_llm/transformers/models/llama.py

```diff
@@ -639,11 +639,19 @@ def llama_attention_forward_4_31_original(
         attn_weights = None
     else:
         # otherwise, use native attention
-        attn_output, attn_weights = native_sdp(query_states, key_states, value_states,
-                                               attention_mask,
-                                               bsz, q_len, kv_seq_len,
-                                               self.head_dim, self.num_heads)
-
+        if not output_attentions and query_states.dtype == torch.float16 and \
+                query_states.shape[2] >= 6800:
```

Comment:
refactor the check into a utility function

Reply:
Sure, fixed.